### PR TITLE
[FLINK-8916] Checkpointing Mode is always shown to be "At Least Once" in Web UI

### PR DESCRIPTION
**Pull request by yanghua**
## What is the purpose of the change

*This pull request add a customized serializer for checkpoint processing mode to support the Web UI's Presentation. *

## Brief change log

  - *Add a customized serializer for checkpoint processing mode*

## Verifying this change

This change is already covered by existing tests*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointConfigInfo.java [hunk 1]
```diff
@@ -143,10 +152,66 @@ public int hashCode() {
...
+	public static class ProcessingModeSerializer extends StdSerializer<ProcessingMode> {
+
+		public ProcessingModeSerializer() {
+			super(ProcessingMode.class);
+		}
+
+		@Override
+		public void serialize(ProcessingMode mode, JsonGenerator generator, SerializerProvider serializerProvider)
+			throws IOException {
+			generator.writeString(mode.getValue());
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointConfigInfo.java]: replace with `generator.writeString(mode.name().toLowerCase())`

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointConfigInfo.java [hunk 2]
```diff
@@ -143,10 +152,66 @@ public int hashCode() {
...
+	public static class ProcessingModeDeserializer extends StdDeserializer<ProcessingMode> {
+
+		public ProcessingModeDeserializer() {
+			super(ProcessingMode.class);
+		}
+
+		@Override
+		public ProcessingMode deserialize(JsonParser jsonParser, DeserializationContext deserializationContext)
+			throws IOException {
+			return ProcessingMode.fromString(jsonParser.getValueAsString());
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointConfigInfo.java]: replace with `ProcessingMode.valueOf(jsonParser.getValueAsString.toUpperCase())`

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointConfigInfo.java [hunk 3]
```diff
@@ -143,10 +152,66 @@ public int hashCode() {
...
+
+		ProcessingMode(String value) {
+			this.value = value;
+		}
+
+		public String getValue() {
+			return value;
+		}
+
+		public static ProcessingMode fromString(String value) {
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointConfigInfo.java]: these changes to the enum seem unnecessary

**Comment by yanghua** [comment]
@zentol thank you for suggestion, please review again.

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**